src/collection.ts

```diff
@@ -1011,6 +1012,57 @@ export class Collection<
     )
   }
 
+  /**
+   * Update the value of one existing documents in the collection.
+   *
+   * @example
+   * ```ts
+   * // Updates the first user document and sets name = 67
+   * await db.users.updateMany({ age: 67 })
+   * ```
+   *
+   * @example
+   * ```ts
+   * // Updates the first user where age > 20, using shallow merge
+   * await db.users.updateOne({ age: 67 }, {
+   *   filter: (doc) => doc.value.age > 20,
+   *   strategy: "merge-shallow"
+   * })
+   * ```
+   *
+   * @example
+   * ```ts
+   * // Finds the first user with username `april`, and sets age to 21
+   * // Attempts to create a user with `age: 21` if it doesn't exist
+   * // (but fails if type validations require any other properties)
+   * const { result } = await db.users.updateOne({ age: 21 }, {
+   *   filter: (doc) => doc.value.username == "april",
+   *   upsert: true
+   * })
+   * ```
+   *
+   * @param value - Updated value to be inserted into documents.
+   * @param options - Update many options, optional.
+   * @returns Promise resolving to an object containining result.
+   */
+  async updateOne(value, options?: UpdateOneOptions<Document<TOutput>>) {
+    const { result: target } = this.getOne(options)
+
+    if (target?.id) {
+      const { result } = await this.update(target.id, value)
+
+      return result
```

Comment:
Here, as with the next return statement a few lines under and for getMany(), I think the return signature makes more sense to be either just the single document (not in an array), or null.